src/screens/DiscoveryScreenV2/index.js

```diff
@@ -69,8 +77,18 @@ const DiscoveryScreenV2 = ({route}) => {
 
   const fetchDiscoveryData = async (text) => {
     const fetchDiscoveryInitialUsers = async () => {
-      const initialData = await DiscoveryRepo.fetchInitialDiscoveryUsers();
-      DiscoveryAction.setDiscoveryInitialUsers(initialData.suggestedUsers, discoveryDispatch);
+      const initialData = await DiscoveryRepo.fetchInitialDiscoveryUsers(
+        50,
+        parseInt(userPage.currentPage, 10)
+      );
+      setUserPage({
+        currentPage: initialData.page,
+        totalPage: initialData.total_page
+      });
+      DiscoveryAction.setDiscoveryInitialUsers(
+        [...discoveryData.initialUsers, ...initialData.suggestedUsers],
+        discoveryDispatch
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for fetching data and updating the state is repeated in multiple places (`fetchDiscoveryInitialUsers`, `fetchDiscoveryInitialDomains`, `fetchDiscoveryInitialTopics`). Consider creating a generic function that takes the necessary parameters and handles this logic to reduce redundancy and improve maintainability.
```diff
- const fetchDiscoveryInitialUsers = async () => {
-   const initialData = await DiscoveryRepo.fetchInitialDiscoveryUsers(
-     50,
-     parseInt(userPage.currentPage, 10)
-   );
-   setUserPage({
-     currentPage: initialData.page,
-     totalPage: initialData.total_page
-   });
-   DiscoveryAction.setDiscoveryInitialUsers(
-     [...discoveryData.initialUsers, ...initialData.suggestedUsers],
-     discoveryDispatch
-   );
- };

+ const fetchDiscoveryInitialData = async (fetchFunction, setPage, setDiscoveryAction, initialDataArray) => {
+   const initialData = await fetchFunction(
+     50,
+     parseInt(userPage.currentPage, 10)
+   );
+   setPage({
+     currentPage: initialData.page,
+     totalPage: initialData.total_page
+   });
+   setDiscoveryAction(
+     [...initialDataArray, ...initialData.suggestedUsers],
+     discoveryDispatch
+   );
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->